fs/sync/sync_test.go

```diff
@@ -1442,6 +1442,38 @@ func TestSyncOverlap(t *testing.T) {
 	checkErr(Sync(ctx, FremoteSync, FremoteSync, false))
 }
 
+// Test a sync with filtered overlap
+func TestSyncOverlapWithFilter(t *testing.T) {
+	ctx := context.Background()
+	r := fstest.NewRun(t)
+	defer r.Finalise()
+
+	fi, err := filter.NewFilter(nil)
+	require.NoError(t, err)
+	require.NoError(t, fi.Add(false, "/rclone-sync-test/"))
+	ctx = filter.ReplaceConfig(ctx, fi)
+
+	subRemoteName := r.FremoteName + "/rclone-sync-test"
+	FremoteSync, err := fs.NewFs(ctx, subRemoteName)
+	FremoteSync.Mkdir(ctx, "")
```

Comment:
The error for this needs to be checked. 

    require.NoError(t, FremoteSync.Mkdir(ctx, ""))

is fine